server/src/main/java/org/opensearch/search/SearchService.java

```diff
@@ -272,6 +272,15 @@ public class SearchService extends AbstractLifecycleComponent implements IndexEv
         Property.NodeScope
     );
 
+    // value 0 means rewrite filters optimization in aggregations will be disabled
+    public static final Setting<Integer> MAX_AGGREGATION_REWRITE_FILTERS = Setting.intSetting(
+        "search.max_aggregation_rewrite_filters",
+        24,
```

Comment:
Why 24?

If we do a daily aggregation for 1 month, would we want the filter rewrite to kick in? If we do an hourly aggregation for 2 days, should we use filter queries or doc values?

Reply:
@bowenlan-amzn - Can we change this to 72 as discussed offline? The value is wide enough to account for some of the cases @msfroh mentioned, but conservative enough to not impact for larger bucket count

Reply:
24 is from the regression of pmc workload, where the optimized method on 24 buckets still perform worse but 12 buckets perform better.
This setting here is easily changeable for open source user and we will have it documented.

Reply:
Updated to 72.